src/main/kotlin/stringcalculator/StringCalculator.kt

```diff
@@ -0,0 +1,28 @@
+package stringcalculator
+
+class StringCalculator(
+    private val validator: Validator,
+    private val numberSumCalculator: NumberSumCalculator,
+    private val stringParser: StringParser
+) {
+
+    fun sum(input: String): Int {
+        if (stringParser.hasCustomDelimiter(input)) {
+            return sumWithCustomDelimiter(input)
+        }
+        return sumWithDefaultDelimiter(input)
+    }
+
+    private fun sumWithDefaultDelimiter(input: String): Int {
+        val target = stringParser.parse(input)
+        validator.ensureAllPositiveNumbers(target)
+        return numberSumCalculator.sum(target.map { it.toInt() })
+    }
+
+    private fun sumWithCustomDelimiter(input: String): Int {
+        val parseDelimiter = stringParser.parseDelimiter(input)
+        val target = stringParser.parse(parseDelimiter.data, parseDelimiter.delimiter)
+        validator.ensureAllPositiveNumbers(target)
+        return numberSumCalculator.sum(target.map { it.toInt() })
+    }
```

Comment:
가급적이면 메서드의 호출순서대로 메서드를 위치시키면 코드를 읽기가 더 쉬운 것 같아요. 😉

```suggestion
    private fun sumWithCustomDelimiter(input: String): Int {
        val parseDelimiter = stringParser.parseDelimiter(input)
        val target = stringParser.parse(parseDelimiter.data, parseDelimiter.delimiter)
        validator.ensureAllPositiveNumbers(target)
        return numberSumCalculator.sum(target.map { it.toInt() })
    }

    private fun sumWithDefaultDelimiter(input: String): Int {
        val target = stringParser.parse(input)
        validator.ensureAllPositiveNumbers(target)
        return numberSumCalculator.sum(target.map { it.toInt() })
    }
```